regression/scripts/release_cray.sh

```diff
@@ -124,7 +98,10 @@ export TARGET=knl
 ## Generic setup
 ##---------------------------------------------------------------------------##
 
-export script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+export script_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" )
```

Comment:
Could this line be `script_dir=$( dirname "${BASH_SOURCE[0]}" )`?
If so, how does line 103 cd into the non-existent directory of `BASH_SOURCE[0]`?
Apologies if I'm missing something.

Reply:
Surprisingly no. The reason is that some 'shortcut' names will not be properly expanded.  For example create `$HOME/foo.sh` with the following content:

```
#!/bin/bash

export script_dir1=$( dirname "${BASH_SOURCE[0]}" )
echo "script_dir1 = $script_dir1"
if ! [[ -d $script_dir1 ]]; then
  export script_dir1="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
fi
echo "script_dir1 = $script_dir1"

export script_dir2=$( cd "$( dirname "${BASH_SOURCE[0]}" )" )
echo "script_dir2 = $script_dir2"
if ! [[ -d $script_dir2 ]]; then
  export script_dir2="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
fi
echo "script_dir2 = $script_dir2"
```

Now try running this script from different locations:

```
% cd $HOME
% ./foo.sh
script_dir1 = .
script_dir1 = .
script_dir2 = 
script_dir2 = /users/kellyt

% cd /tmp
% $HOME/foo.sh
script_dir1 = /users/kellyt
script_dir1 = /users/kellyt
script_dir2 = 
script_dir2 = /users/kellyt

% cd $HOME/..
% $LOGNAME/foo.sh
script_dir1 = kellyt
script_dir1 = kellyt
script_dir2 = 
script_dir2 = /users/kellyt
```